File: app/Models/Cliente.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Cliente extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'nombre',
        'documento',
        'telefono',
        'email',
        'direccion',
        'notas',
        'activo',
        'created_by',
    ];

    protected function casts(): array
    {
        return [
            'activo' => 'boolean',
        ];
    }

    public function creador()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function cotizaciones()
    {
        return $this->hasMany(Cotizacion::class);
    }

    public function facturas()
    {
        return $this->hasMany(Factura::class);
    }
}

File: mobile_app/lib/features/clientes/domain/cliente.dart
class Cliente {
  Cliente({
    required this.id,
    required this.nombre,
    required this.documento,
    required this.telefono,
    required this.email,
    required this.activo,
  });

  final int id;
  final String nombre;
  final String documento;
  final String telefono;
  final String email;
  final bool activo;

  factory Cliente.fromJson(Map<String, dynamic> json) {
    return Cliente(
      id: json['id'] as int,
      nombre: json['nombre']?.toString() ?? '',
      documento: json['documento']?.toString() ?? '',
      telefono: json['telefono']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      activo: json['activo'] as bool? ?? false,
    );
  }
}
